2.18 What's new and Language evolution pages (#4195)
dart-lang/site-www: src/_guides/language/evolution.md

> ### Dart 2.18

Reviewer: ```suggestion
### August 31, 2022: Dart 2.18 release
```

> Dart 2.18 enhanced type inference by allowing information flow between arguments in generic function calls.
> Leaving the argument type unspecified within some invocations previously triggered type errors,
> citing potential null occurrences.
> The compiler can now infer the argument type from other values in an invocation,
> so specifying the argument type inline is no longer necessary. 
> For more information,
> see [Type argument inference][].

Reviewer 2: **suggestion:** Simplify the copy. Sometimes, it's better to cut sentences into shorter, linked ideas.
```suggestion
Dart 2.18 enhanced type inference. This change allows information flow between arguments in generic function calls. Before 2.18, if you didn't specify an argument's type in some methods, Dart returned errors. These type errors cited potential null occurrences. With 2.18, the compiler infers the argument type from other values in an invocation. You don't need to specify the argument type inline. To learn more, see [Type argument inference][].
```

As written, this is Grade 14 readability at 63 words. As revised, it's Grade 7 with 71 words. Clear, concise copy should be the goal. 

**issue:** This may be personal preference, but I really have an issue with "For more information..." or "For more details...". This phrasing removes agency from the reader. I prefer "To learn more...". This serves as a call to action.

> [Type system]: /guides/language/type-system

Author: @mit-mit would the [Type argument inference](https://dart.dev/guides/language/type-system#type-argument-inference) section specifically be a better place to link to for more information on this topic?

If so, could you please confirm none of the description there needs to be updated due to this enhancement?

Reviewer 3: Yes, that would be a fine place for it

> see the [Type system][] page.
> 
> [Type system]: /guides/language/type-system#type-argument-inference

Reviewer 4: Consider being more specific here:

```suggestion
see [Type argument inference][].

[Type argument inference]: /guides/language/type-system#type-argument-inference
```
